src/mappings/gns.ts

```diff
@@ -328,9 +394,54 @@ export function handleNameSignalUpgrade(event: NameSignalUpgrade): void {
   // Weirdly here, we add the token amount to both, but also the name curator owner must
   // stake the withdrawal fees, so both balance fairly
   // TODO - will have to come back here and make sure my thinking is correct
+  // event.params.newVSignalCreated -> will be used to calculate new nSignal/vSignal ratio
+  subgraph.signalAmount = event.params.newVSignalCreated
   subgraph.unsignalledTokens = subgraph.unsignalledTokens.plus(event.params.tokensSignalled)
   subgraph.signalledTokens = subgraph.signalledTokens.plus(event.params.tokensSignalled)
   subgraph.save()
+
+  let signalRatio = subgraph.signalAmount / subgraph.nameSignalAmount
```

Comment:
I believe we will have to update both of these types to BigDecimal. We can get in a scenario where the new signal amount is less than the name signal amount. If that happened, division of two big ints would just give us zero

Reply:
Hmm, you are right, I forgot the `/` returned a BigInt 🤦🏼